forch/faucet_state_collector.py

```diff
@@ -569,24 +571,71 @@ def _fill_path_to_root(self, switch_name, switch_map):
         """populate path to root for switch_state"""
         switch_map["root_path"] = self.get_switch_egress_path(switch_name)
 
-    def _fill_dva_states(self, switch_name, port_id, port_map):
-        dp_obj = self.faucet_config.get(DPS_CFG, {}).get(switch_name)
-        if not dp_obj:
+    def _fill_vlan_behavior(self, switch_name, switch_map, metrics=None):
+        dp_config = self.faucet_config.get(DPS_CFG, {}).get(switch_name)
+        if not dp_config:
             LOGGER.warning('Switch not defined in dps config: %s', switch_name)
             return
 
-        port_obj = dp_obj.interfaces.get(port_id)
-        if not port_obj:
+        vlans_map = switch_map.setdefault('vlans', {})
+        for vid, vlan_config in dp_config.vlans.items():
+            acl_maps_list = vlans_map.setdefault(int(vid), {}).setdefault('acls', [])
+            if metrics and 'flow_packet_count_vlan_acl' in metrics:
```

Comment:
this seems fragile -- under what conditions would 'flow_packet_cound_vlan_acl' not be in metrics?

Reply:
In the test I have seen the VLAN acl metric is always there, but not the port ACL. Still investigating in which condition a metric will be created.

Reply:
It might be a startup race condition...?

Reply:
Ok, finally found the problem.. Gauge is not configured to be auto reload faucet.yaml. After adding GAUGE_CONFIG_STAT_RELOAD=1 the port acl metric is added.

Back to your question, what I observed is that if no ACL is defined for a VLAN then there there will be no metrics for this VLAN. Up to this line in the code, we cannot gurantee that there are any ACLs defined for this VLAN.

Btw, I took out the sample assignmet out of the vlans loop.